File: app/src/main/java/vasuki/istanpdf/domain/ReorderPdf.java
package vasuki.istanpdf.domain;

import android.net.Uri;

import java.util.List;

import vasuki.istanpdf.data.PdfEngine;
import vasuki.istanpdf.model.PageItem;

public final class ReorderPdf {
    private final PdfEngine pdfEngine;

    public ReorderPdf(PdfEngine pdfEngine) {
        this.pdfEngine = pdfEngine;
    }

    public int execute(Uri source, List<PageItem> pages, Uri destination) throws Exception {
        if (source == null) {
            throw new IllegalArgumentException("Source PDF is required");
        }
        return pdfEngine.reorder(source, pages, destination);
    }
}

File: app/src/main/kotlin/vasuki/istanpdf/domain/ReorderPdf.kt
package vasuki.istanpdf.domain

import android.net.Uri
import vasuki.istanpdf.data.PdfEngine
import vasuki.istanpdf.model.PageItem

class ReorderPdf(private val pdfEngine: PdfEngine) {
    @Throws(Exception::class)
    fun execute(source: Uri, pages: List<PageItem>, destination: Uri): Int {
        return pdfEngine.reorder(source, pages, destination)
    }
}
